File: backend/models/article.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Article struct {
	ID uint `gorm:"primaryKey" json:"id"`

	// Relations
	CategoryID uint     `gorm:"not null;index" json:"category_id"`
	Category   Category `gorm:"foreignKey:CategoryID" json:"category"`

	Tags []Tag `gorm:"many2many:article_tags;" json:"tags"`

	// Content
	Title string `gorm:"size:255;not null" json:"title"`

	Slug string `gorm:"size:255;uniqueIndex;not null" json:"slug"`

	Excerpt string `gorm:"type:text" json:"excerpt"`

	ContentMarkdown string `gorm:"type:longtext" json:"content_markdown"`

	ContentHTML string `gorm:"type:longtext" json:"content_html"`

	FeaturedImage string `gorm:"size:500" json:"featured_image"`

	// SEO
	SeoTitle string `gorm:"size:70" json:"seo_title"`

	SeoDescription string `gorm:"size:160" json:"seo_description"`

	CanonicalURL string `gorm:"size:500" json:"canonical_url"`

	MetaKeywords string `gorm:"size:255" json:"meta_keywords"`

	Robots string `gorm:"size:30;default:index,follow" json:"robots"`

	// Publish
	Status string `gorm:"type:enum('draft','published','scheduled');default:'draft'" json:"status"`

	IsFeatured bool `gorm:"default:false" json:"is_featured"`

	PublishedAt *time.Time `json:"published_at"`

	// Analytics
	ViewCount int64 `gorm:"default:0" json:"view_count"`

	ReadingTime int `gorm:"default:1" json:"reading_time"`

	// Audit
	CreatedAt time.Time `json:"created_at"`

	UpdatedAt time.Time `json:"updated_at"`

	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`
}

File: admin/types/article.ts
import { Category } from "./category";
import { Tool } from "./tool";
import { Media } from "./media";

export interface ArticleForm {
    title: string;
    slug: string;
    category_id: number;
    primary_tool_id?: number | null;
    featured_image_id?: number | null;
    tags: number[];
    excerpt: string;
    content_markdown: string;
    featured_image: string;
    seo_title: string;
    seo_description: string;
    canonical_url: string;
    status: "draft" | "published";
    is_featured: boolean;
}

export interface Article {
    id: number;
    title: string;
    slug: string;
    category_id?: number;
    category?: Category;
    primary_tool_id?: number | null;
    primary_tool?: Tool | null;
    primaryTool?: Tool | null;
    featured_image_id?: number | null;
    featured_image_media?: Media | null;
    featuredImageMedia?: Media | null;
    tags: number[];
    excerpt: string;
    content_markdown: string;
    featured_image: string;
    seo_title: string;
    seo_description: string;
    canonical_url: string;
    status: "draft" | "published";
    is_featured: boolean;
    published_at: string;
    created_at: string;
    updated_at: string;
    view_count: number;
    reading_time: number;
}

export interface ArticleResponse {
    success: boolean;
    message: string;
    data: Article[];
    meta: {
        page: number;
        limit: number;
        total: number;
    };
}